src/ccpp_scheme_utils.F90

```diff
@@ -0,0 +1,101 @@
+module ccpp_scheme_utils
+
+   ! Module of utilities available to CCPP schemes
+
+   use ccpp_constituent_prop_mod, only: ccpp_model_constituents_t, int_unassigned
+
+   implicit none
+   private
+
+   !! Public interfaces
+   public :: ccpp_initialize_constituent_ptr ! Used by framework to initialize
+   public :: ccpp_constituent_index          ! Lookup index constituent by name
+   public :: ccpp_constituent_indices        ! Lookup indices of consitutents by name
+
+   !! Private module variables & interfaces
+
+   ! initialized set to .true. once hash table pointer is initialized
+   logical                                  :: initialized = .false.
+   type(ccpp_model_constituents_t), pointer :: constituent_obj => NULL()
+
+   private :: uninitialized
+
+contains
+
+   subroutine uninitialized(caller, errcode, errmsg)
+      ! Dummy arguments
+      character(len=*),           intent(in)  :: caller
+      integer,          optional, intent(out) :: errcode
+      character(len=*), optional, intent(out) :: errmsg
+
+      if (.not. initialized) then
+         if (present(errcode)) then
+            errcode = 1
+         end if
+         if (present(errmsg)) then
+            errmsg = trim(caller)//' FAILED, module not initialized'
+         end if
+      end if
+   end subroutine uninitialized
+
+   subroutine ccpp_initialize_constituent_ptr(const_obj)
+      ! Dummy arguments
+      type(ccpp_model_constituents_t), pointer, intent(in) :: const_obj
+
+      if (.not. initialized) then
+         constituent_obj => const_obj
+         initialized = .true.
+      end if
+   end subroutine ccpp_initialize_constituent_ptr
+
+   subroutine ccpp_constituent_index(standard_name, const_index, errcode, errmsg)
```

Comment:
General design question - what is the anticipated usage of these interfaces on the physics side? 

As written, the code would return `int_unassigned` for the index if the provided constituent isn't present. So would it be expected that the physics would check for `int_unassigned` after calling `ccpp_constituent_index`?

Reply:
@gold2718 @peverwhee I haven't thought through how we would tackle implementing the constituents feature in the UFS, so I don't have a strong opinion. On the physics side, having to check that `index ne int_unassigned` before using index seems sensible to me. At least with my (limited) understanding.

Reply:
I think this makes sense. Someone may want to iterate over constituents to see what is available and what not. That is, not every call to this scheme should result in errflg /= success.